src/SoundOptionsModal.js

```diff
@@ -0,0 +1,145 @@
+// @flow
+
+import React from 'react';
+import classNames from 'classnames';
+import ModalBody from './ModalBody';
+import ModalHeader from './ModalHeader';
+import ModalWithFooter from './ModalWithFooter';
+import ToggleSwitch from './ToggleSwitch';
+import { injectIntl } from 'react-intl';
+import type {IntlShape} from 'react-intl';
+import { ReactComponent as AudioIcon } from './svg/AudioOn.svg';
+import './SoundOptionsModal.scss';
+
+type SoundOptionsModalProps = {
+    intl: IntlShape,
+    audioEnabled: boolean,
+    announcementsEnabled: boolean,
+    sonificationEnabled: boolean,
+    show: boolean,
+    onCancel: () => void,
+    onChangeSoundOptions: ( audioEnabled: boolean, announcementsEnabled: boolean, sonificationEnabled: boolean ) => void
+};
+
+type SoundOptionsModalStates = {
+    audioEnabled: boolean,
+    announcementsEnabled: boolean,
+    sonificationEnabled: boolean
+};
+
+class SoundOptionsModal extends React.Component<SoundOptionsModalProps, SoundOptionsModalStates> {
+    constructor (props: SoundOptionsModalProps) {
+        super(props);
+        this.state = {
+            audioEnabled: this.props.audioEnabled,
+            announcementsEnabled: this.props.announcementsEnabled,
+            sonificationEnabled: this.props.sonificationEnabled
+        };
+    };
+
+    handleToggleAudioEnabled = () => {
+        this.setState({
+            audioEnabled: !this.state.audioEnabled
+        });
+    };
+
+    handleToggleAnnouncementsEnabled = () => {
+        this.setState({
+            announcementsEnabled: !this.state.announcementsEnabled
+        });
+    };
+
+    handleToggleSonificationEnabled = () => {
+        this.setState({
+            sonificationEnabled: !this.state.sonificationEnabled
+        });
+    };
+
+    handleOnSave = () => {
+        this.props.onChangeSoundOptions(this.state.audioEnabled, this.state.announcementsEnabled, this.state.sonificationEnabled );
+    }
+
+    handleOnCancel = () => {
+        this.setState({
+            audioEnabled: this.props.audioEnabled,
+            announcementsEnabled: this.props.announcementsEnabled,
+            sonificationEnabled: this.props.sonificationEnabled
+        });
```

Comment:
Rather than setting the state to the props values at cancel, we need to set them at show. See my comment on `SoundOptionsModal.test.js` which goes into more detail and illustrates a bug with this approach.